src/java/org/apache/cassandra/db/marshal/AbstractTimeUUIDType.java

```diff
@@ -83,16 +119,30 @@ public static long signedBytesToNativeLong(long signedBytes)
         return signedBytes ^ 0x0080808080808080L;
     }
 
-    private static long topbyte(long topbyte)
+    private void verifyVersion(long hiBits)
     {
-        return topbyte << 56;
+        long version = (hiBits >>> 12) & 0xf;
+        if (version != 1)
+            throw new MarshalException(String.format("Invalid UUID version %d for timeuuid",
```

Comment:
nit: untested codepath (although there's something close in `TypeValidationTest` I think)